src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobMetadataStore.cs

```diff
@@ -116,25 +118,50 @@ await using (Utf8JsonWriter utf8Writer = new Utf8JsonWriter(stream))
     public async Task DeleteInstanceMetadataIfExistsAsync(VersionedInstanceIdentifier versionedInstanceIdentifier, CancellationToken cancellationToken)
     {
         EnsureArg.IsNotNull(versionedInstanceIdentifier, nameof(versionedInstanceIdentifier));
-        BlockBlobClient[] blobClients = GetInstanceBlockBlobClients(versionedInstanceIdentifier);
 
-        await Task.WhenAll(blobClients.Select(blob => ExecuteAsync(t => blob.DeleteIfExistsAsync(DeleteSnapshotsOption.IncludeSnapshots, conditions: null, t), cancellationToken)));
+        try
+        {
+            BlockBlobClient[] blobClients = GetInstanceBlockBlobClients(versionedInstanceIdentifier);
+
+            await Task.WhenAll(blobClients.Select(blob => ExecuteAsync(t => blob.DeleteIfExistsAsync(DeleteSnapshotsOption.IncludeSnapshots, conditions: null, t), cancellationToken)));
+        }
+        catch (Exception ex)
+        {
+            _logger.LogWarning(ex, "The operation failed.");
+
+            throw;
+        }
     }
 
     /// <inheritdoc />
     public Task<DicomDataset> GetInstanceMetadataAsync(VersionedInstanceIdentifier versionedInstanceIdentifier, CancellationToken cancellationToken)
     {
         EnsureArg.IsNotNull(versionedInstanceIdentifier, nameof(versionedInstanceIdentifier));
-        BlockBlobClient blobClient = GetInstanceBlockBlobClient(versionedInstanceIdentifier, _blobMigrationFormatType);
 
-        return ExecuteAsync(async t =>
+        try
         {
-            BlobDownloadResult result = await blobClient.DownloadContentAsync(t);
+            BlockBlobClient blobClient = GetInstanceBlockBlobClient(versionedInstanceIdentifier, _blobMigrationFormatType);
+            return ExecuteAsync(async t =>
+            {
+                BlobDownloadResult result = await blobClient.DownloadContentAsync(t);
 
-            // DICOM metadata file includes UTF-8 encoding with BOM and there is a bug with the BinaryData.ToObjectFromJson method as seen in this issue: https://github.com/dotnet/runtime/issues/71447
-            // So passing as stream which will remove the UTF-8 BOM.
-            return await JsonSerializer.DeserializeAsync<DicomDataset>(result.Content.ToStream(), _jsonSerializerOptions, t);
-        }, cancellationToken);
+                // DICOM metadata file includes UTF-8 encoding with BOM and there is a bug with the BinaryData.ToObjectFromJson method as seen in this issue: https://github.com/dotnet/runtime/issues/71447
+                // So passing as stream which will remove the UTF-8 BOM.
+                return await JsonSerializer.DeserializeAsync<DicomDataset>(result.Content.ToStream(), _jsonSerializerOptions, t);
+            }, cancellationToken);
+        }
+        catch (ItemNotFoundException ex)
```

Comment:
Where is `ItemNotFoundException` from?

Reply:
We throw it when attempting to retrieve a blob and `catch (RequestFailedException ex) when (ex.ErrorCode == BlobErrorCode.BlobNotFound)`